src/app.js

```diff
@@ -274,7 +267,7 @@ class App {
     this.merge = new Merge(this.pouch)
     this.prep = new Prep(this.merge, this.ignore)
     this.local = this.merge.local = new Local(this.config, this.prep, this.pouch, this.events)
-    this.remote = this.merge.remote = new Remote(this.config, this.prep, this.pouch, this.events)
+    this.remote = this.merge.remote = new Remote(this.config)
```

Comment:
Because there is a design issue (cyclic dependency) in the existing v2 code: `Prep -> Merge -> Remote -> Prep`.

Since refactoring all those components was out of scope, I didn't change the assignment.